core/services/directrequestocr/drlistener.go

```diff
@@ -23,6 +25,56 @@ import (
 var (
 	_ log.Listener   = &DRListener{}
 	_ job.ServiceCtx = &DRListener{}
+
+	promOracleEvents = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "functions_oracle_events",
+		Help: "Metric to track oracle events",
+	}, []string{"oracle", "event"})
+
+	promRequestInternalErrors = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "functions_request_internal_errors",
+		Help: "Metric to track internal errors",
+	}, []string{"oracle"})
+
+	promRequestComputationErrors = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "functions_request_computation_errors",
+		Help: "Metric to track computation errors",
+	}, []string{"oracle"})
+
+	promRequestComputationSuccess = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "functions_request_computation_success",
+		Help: "Metric to track number of computed requests",
+	}, []string{"oracle"})
+
+	promRequestTimeout = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "functions_request_timeout",
+		Help: "Metric to track number of timed out requests",
+	}, []string{"oracle"})
+
+	promRequestConfirmed = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "functions_request_confirmed",
+		Help: "Metric to track number of confirmed requests",
+	}, []string{"oracle", "responseType"})
+
+	promRequestDataSize = promauto.NewGaugeVec(prometheus.GaugeOpts{
```

Comment:
I'm not sure if these last 4 counters here make sense. IIUC they will accumulate a sum of all sizes/durations of all requests. How can we use that data? To get an average? I guess that might make sense for duration (although a p95/p99 would be more useful) but not so much for sizes IMO. Thoughts?

Reply:
Now using Histogram.